app/src/main/java/com/nextcloud/talk/ui/WaveformSeekBar.kt

```diff
@@ -0,0 +1,118 @@
+ */
+
+package com.nextcloud.talk.ui
+
+import android.content.Context
+import android.content.res.Resources
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.util.AttributeSet
+import androidx.annotation.ColorInt
+import androidx.appcompat.widget.AppCompatSeekBar
+import kotlin.math.roundToInt
+
+class WaveformSeekBar : AppCompatSeekBar {
+
+    @ColorInt
+    private var primary: Int = Color.parseColor("#679ff5")
+
+    @ColorInt
+    private var secondary: Int = Color.parseColor("#a6c6f7")
+    private var waveData: FloatArray = floatArrayOf()
+    private val paint: Paint = Paint(Paint.ANTI_ALIAS_FLAG)
+
+    constructor(context: Context) : super(context) {
+        init()
+    }
+
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
+        init()
+    }
+
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
+        init()
+    }
+
+    fun setColors(@ColorInt p: Int, @ColorInt s: Int) {
+        primary = p
+        secondary = s
+        invalidate()
+    }
+
+    fun setWaveData(data: FloatArray) {
+        waveData = data
+        invalidate()
+    }
+
+    private fun init() {
+        paint.apply {
+            strokeCap = Paint.Cap.ROUND
+            strokeWidth = DEFAULT_BAR_WIDTH.dp.toFloat()
+            color = Color.RED
+        }
+    }
+
+
+    override fun onDraw(canvas: Canvas?) {
+        if (waveData.size == 0 || waveData[0].toString() == "NaN") {
+            super.onDraw(canvas)
+        } else {
+            if (progressDrawable != null) {
+                super.setProgressDrawable(null)
+            }
+
+            drawWaveformSeekbar(canvas)
+            super.onDraw(canvas)
+        }
+    }
+
+    private fun drawWaveformSeekbar(canvas: Canvas?) {
+        val usableHeight = height - paddingTop - paddingBottom
+        val usableWidth = width - paddingLeft - paddingRight
+        val midpoint = usableHeight / 2f
+        val maxHeight: Float = usableHeight / MAX_HEIGHT_DIVISOR
+        val barGap: Float = (usableWidth - waveData.size * DEFAULT_BAR_WIDTH) / (waveData.size - 1).toFloat()
+
+        canvas?.apply {
+            save()
+            translate(paddingLeft.toFloat(), paddingTop.toFloat())
+            for (i in waveData.indices) {
+                val x: Float = i * (DEFAULT_BAR_WIDTH + barGap) + DEFAULT_BAR_WIDTH / 2f
+                val y: Float = waveData[i] * maxHeight
+                val progress = (x / usableWidth)
+                paint.color = if (progress * max < getProgress()) primary else secondary
+                canvas.drawLine(x, midpoint - y, x, midpoint + y, paint)
+            }
+
+            restore()
+
+        }
+    }
+
+    companion object {
+        private val TAG: String? = WaveformSeekBar::class.simpleName
+        private const val DEFAULT_BAR_WIDTH: Int = 5
```

Comment:
If we change this to 2, we get many more "bars"? I would prefer to have much more, on your video it looks like we have 20 bars while the telegram screenshot is around 50 or something?

Reply:
No problem, how's this? 
<img width="400" alt="image" src="https://github.com/nextcloud/talk-android/assets/69230048/7c3f6d87-8261-459c-89d3-7f8c34207353">


I set the bar width to 1dp and made the bars 50. While the bar width is const, the number of bars is set by the `shrinkFloatArray` function in `AudioUtils`. The function `audioFileToFloatArray` usually returns a large array with hundreds of values, so I created this helper function to shrink it. This code is located on line 874 in `setWaveform` in `ChatActivity`
```
        val r = AudioUtils.audioFileToFloatArray(file)
        val f = AudioUtils.shrinkFloatArray(r, 50)
```

Reply:
I would suggest slightly wider and also higher -comparing it to the seekbar's "nob" if that is doable

Reply:
![image](https://github.com/nextcloud/talk-android/assets/69230048/493b18b8-2187-43ee-bb0d-d150ff5d30dc)

I made ```MAX_HEIGHT_DIVISOR: Float = 10f``` to ```MAX_HEIGHT_DIVISOR: Float = 6.5f```